thereseLYR/keybored-proto
# feat: add user endpoints for login + register

## Changes
- Add endpoint for logging a user in 
- Add endpoint for creating a new user
- Some db changes with user models (TODO: might need to add foreign key relation to songs table in future)
- Made some minor refactoring to routes

## How to test
Before testing the endpoint, you might want to run the migrations again
```
npx sequelize db:migrate:undo:all
npx sequelize db:migrate
```
- Add SALT value to .env file
- Test endpoint locally with thunder client/ postman
- Send POST request to http://localhost:3004/register with a username and password payload, expect to receive 200 with a new user created in db
- Send POST request to http://localhost:3004/session with new user created username and password payload, expect to receive 200 with a login successful message with hash token


## Screenshots (if any)
/session endpoint
<img width="1132" alt="image" src="https://user-images.githubusercontent.com/30279812/184460951-0d634f3b-c9e1-4d6c-8bef-8d9fc1246f95.png">
/register endpoint
<img width="1130" alt="image" src="https://user-images.githubusercontent.com/30279812/184460973-6f87f40d-6917-476b-bd47-eb7cfbc67f45.png">
